### PR TITLE
Duckdb wasm loadable fixes

### PR DESCRIPTION
**Pull request by carlopi**
Refactor to extension loading to handle duckdb-wasm loading logic

### CODE REVIEW

File: src/main/extension/extension_install.cpp [hunk 1]
```diff
@@ -45,7 +45,9 @@ const vector<string> ExtensionHelper::PathComponents() {
 
 string ExtensionHelper::ExtensionDirectory(DBConfig &config, FileSystem &fs) {
 #ifdef WASM_LOADABLE_EXTENSIONS
-	static_assertion(0, "ExtensionDirectory functionality is not supported in duckdb-wasm");
+#ExtensionDirectory functionality is not supported in duckdb - wasm
```
Comment (Mytherin) [inline src/main/extension/extension_install.cpp:48]: Should this not stay as a static assert? - looks like this was changed accidentally

File: src/main/extension/extension_load.cpp [hunk 2]
```diff
@@ -58,7 +58,7 @@ static void ComputeSHA256FileSegment(FileHandle *handle, const idx_t start, cons
 #endif
 
 bool ExtensionHelper::TryInitialLoad(DBConfig &config, FileSystem &fs, const string &extension,
-                                     ExtensionInitResult &result, string &error) {
+                                     ExtensionInitResult &result, string &error, const ClientConfig *client_config) {
```
Comment (Mytherin) [inline src/main/extension/extension_load.cpp:61]: nit: `optional_ptr`

File: src/main/extension/extension_install.cpp [hunk 3]
```diff
@@ -157,6 +159,32 @@ void WriteExtensionFileToDisk(FileSystem &fs, const string &path, void *data, id
 	target_file.reset();
 }
 
+string ExtensionHelper::ExtensionUrlTemplate(ClientConfig *client_config, const string &repository) {
```
Comment (Mytherin) [inline src/main/extension/extension_install.cpp:162]: nit: `optional_ptr`

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! LGTM - some comments:

**Comment by carlopi** [comment]
Thanks for the review, updated according to the comments!

File: src/include/duckdb/main/extension_helper.hpp [hunk 4]
```diff
@@ -46,13 +46,16 @@ class ExtensionHelper {
 	static void InstallExtension(DBConfig &config, FileSystem &fs, const string &extension, bool force_install,
 	                             const string &respository = "");
 	static void LoadExternalExtension(ClientContext &context, const string &extension);
-	static void LoadExternalExtension(DatabaseInstance &db, FileSystem &fs, const string &extension);
+	static void LoadExternalExtension(DatabaseInstance &db, FileSystem &fs, const string &extension,
+	                                  const ClientConfig *client_config);
```
Comment (Mytherin) [inline src/include/duckdb/main/extension_helper.hpp:50]: nit: `optional_ptr`

**Review by Mytherin — COMMENTED** [review]
Thanks! LGTM - only one more raw pointer remaining

**Comment by carlopi** [comment]
Failure looks to me unrelated to this changes.

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**